src/EnsembleKalmanInversion.jl

```diff
@@ -80,10 +158,28 @@ function update_ensemble!(
     # G is deterministic
     y = deterministic_forward_map ? (ekp.obs_mean .+ noise) : (ekp.obs_mean .+ zero(noise))
 
-    # N_obs × N_obs \ [N_obs × N_ens]
-    # --> tmp is [N_obs × N_ens]
-    tmp = (cov_gg + scaled_obs_noise_cov) \ (y - g)
-    u += (cov_ug * tmp) # [N_par × N_ens]
+    if isnothing(failure_handler)
```

Comment:
Rather than the if-else's and strings. It might be nice to wrap up error handling a little more, I was wondering about floating the idea of this:
```julia
struct FailureHandler{Process::P}
    update_failure_method::Function
end
function GaussianFailure(Process::P)
    update_failure_method = ...
    return FailureHandler{P}(update_failure_method)
end
```
then have in the update we call  `handle_failures(u,g,y,cov,failure_handler)` that does something like this
```julia 
function handle_failures(u,g,y,cov,failure_handler)
    failure_handler.update_failure_method(u,g,y,cov)
end
```
Or something else, I'm not quite sure yet what would be smoothest across the different methods but this sort of thing could be nice! Any thoughts?

The benefit here would be that someone can add new failure methods without tampering with the updates, or if they want parameters in the failure methods, this will not change the input arguments to the update_ensemble.